### ISSUE TITLE
Pattern matching within a lambda within a field initializer (and IDE0019)

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#18671]
**Version Used**: VS 15.1 (26403.3)

Following code does not compile with error "CS8200	Out variable and pattern variable declarations are not allowed within constructor initializers, field initializers, or property initializers." but i believe it should:
```C#
class Class1
{
    private static readonly Func<object, string> castToString =
        obj =>
        {
            if (obj is string s) return s;
            else return string.Empty;
        };
}
```

There is inconsistency between this behavior and IDE0019 (Use pattern matching). IDE0019 is detected in following code:
```C#
class Class1
{
    private static readonly Func<object, string> castToString =
        obj =>
        {
            var s = obj as string;
            if (s != null) return s;
            else return string.Empty;
        };
}
```
and fix of it looks like a code at the beginning.

Maybe it is a same problem as with #16935 but I'm not sure.

**Comment by chucker** [comment]
This also affects out variables:

```csharp
        private static readonly Lazy<bool> _SomeBool =
            new Lazy<bool>(() =>
            {
                bool result;
                return DBSettings.TryGet("SomeBool", out result) && result;
            });
```

VS will suggest:

```csharp
        private static readonly Lazy<bool> _SomeBool =
            new Lazy<bool>(() =>
            {
                return DBSettings.TryGet("SomeBool", out bool result) && result;
            });
```

But then fail with CS8200. I'm not sure if this is also IDE0019.

**Comment by chucker** [comment]
Possibly a duplicate of #17371

**Comment by gafter** [comment]
This is a symptom of https://github.com/dotnet/roslyn/issues/16935 .
